presto-main/src/main/java/com/facebook/presto/sql/planner/plan/SemiJoinNode.java

```diff
@@ -56,11 +59,22 @@ public SemiJoinNode(@JsonProperty("id") PlanNodeId id,
         this.semiJoinOutput = requireNonNull(semiJoinOutput, "semiJoinOutput is null");
         this.sourceHashSymbol = requireNonNull(sourceHashSymbol, "sourceHashSymbol is null");
         this.filteringSourceHashSymbol = requireNonNull(filteringSourceHashSymbol, "filteringSourceHashSymbol is null");
+        this.method = requireNonNull(method, "method is null");
 
         checkArgument(source.getOutputSymbols().contains(sourceJoinSymbol), "Source does not contain join symbol");
         checkArgument(filteringSource.getOutputSymbols().contains(filteringSourceJoinSymbol), "Filtering source does not contain filtering join symbol");
     }
 
+    public enum Method{
+        DISTRIBUTED,
+        BROADCAST;
```

Comment:
drop unnecessary semicolon and enable IDE warning for that if you have it disabled .